src/ophys_etl/modules/segmentation/postprocess_utils/roi_merging.py

```diff
@@ -0,0 +1,999 @@
+
+    pair_list = [pair for pair in output_list]
+    return pair_list
+
+
+def get_inactive_mask(img_data_shape: Tuple[int, int],
+                      roi_list: List[OphysROI]) -> np.ndarray:
+    """
+    Return a pixel mask that is marked False for all
+    pixels that are part of an ROI, and True for all
+    other pixels
+
+    Parameters
+    ----------
+    img_data_shape: Tuple[int, int]
+        Shape of the full field of view
+
+    roi_list: List[OphysROI]
+
+    Returns
+    -------
+    inactive_pixel_mask: np.ndarray
+    """
+    full_mask = np.zeros(img_data_shape, dtype=bool)
+    for roi in roi_list:
+        mask = roi.mask_matrix
+        region = full_mask[roi.y0:roi.y0+roi.height,
+                           roi.x0:roi.x0+roi.width]
+        full_mask[roi.y0:roi.y0+roi.height,
+                  roi.x0:roi.x0+roi.width] = np.logical_or(mask, region)
+
+    return np.logical_not(full_mask)
+
+
+def get_inactive_dist(img_data: np.ndarray,
+                      roi: OphysROI,
+                      inactive_mask: np.ndarray,
+                      dx: int = 10) -> Tuple[float, float]:
+    """
+    Given an ROI, an array of image data, and a mask marked True at
+    all of the non-ROI pixels, return the mean and standard deviation
+    of the inactive pixels in a neighborhood about the ROI.
+
+    Parameters
+    ----------
+    img_data: np.ndarray
+        The full field of view image data
+
+    roi: OphysROI
+
+    inactive_mask: np.ndarray
+        An array of booleans the same shape as img_data.
+        Marked True for any pixel not in an ROI; False
+        for all other pixels.
+
+    dx: int
+        Number of pixels to either side of the ROI to use
+        when constructing the neighborhood.
+
+    Returns
+    -------
+    mu: float
+        Mean of the inactive pixels in the neighborhood
+
+    std: float
+        Standard deviation of the inactive pixels in the
+        neighborhood
+    """
+
+    xmin = max(0, roi.x0-dx)
+    ymin = max(0, roi.y0-dx)
+    xmax = xmin+roi.width+2*dx
+    ymax = ymin+roi.height+2*dx
+
+    if xmax > img_data.shape[1]:
+        xmax = img_data.shape[1]
+        xmin = max(0, xmax-roi.width-2*dx)
+    if ymax > img_data.shape[1]:
+        ymin = max(0, ymax-roi.height-2*dx)
+
+    neighborhood = img_data[ymin:ymax, xmin:xmax]
+    mask = inactive_mask[ymin:ymax, xmin:xmax]
+    inactive_pixels = neighborhood[mask].flatten()
+    mu = np.mean(inactive_pixels)
+    if len(inactive_pixels) < 2:
+        std = 0.0
+    else:
+        std = np.std(inactive_pixels, ddof=1)
+
+    return mu, std
+
+
+def merge_segmentation_rois(uphill_roi: SegmentationROI,
+                            downhill_roi: SegmentationROI,
+                            new_roi_id: int,
+                            new_flux_value: float) -> SegmentationROI:
+    """
+    Merge two SegmentationROIs, making sure that the ROIs actually
+    abut and that the uphill ROI has a larger flux value than the
```

Comment:
can you explain why this is a requirement, about the flux?
maybe not now, but it seems like the act of merging should be commutative.
i.e.
```
should_merge(A, B) == should_merge(B, A)
```